proto/rill/admin/v1/api.proto

```diff
@@ -1839,6 +1840,14 @@ message GetBillingSubscriptionResponse {
   Subscription subscription = 2;
   string billing_portal_url = 3;
   bool has_payment_method = 4;
+  PaymentCardStatus payment_card_status = 5;
+
+}
+
+enum PaymentCardStatus {
+  PAYMENT_CARD_STATUS_UNSPECIFIED = 0;
+  PAYMENT_CARD_STATUS_OK = 1;
+  PAYMENT_CARD_STATUS_EXPIRED = 2;
 }
```

Comment:
What is the goal of exposing this? Just wondering if we could just track whether they have a valid payment method or not, and direct them to the Stripe portal if they want more details?

As stated elsewhere, I'm also guessing there are many other conditions that could cause a card to be invalid than just expiry.

Reply:
using the api we can just check if there exists a payment method or not for the customer, does explicitly say if its a valid or not.

Reply:
The goal is Eric O mentioned that we want to show a banner if the card expires.

Reply:
Okay, maybe not completely related, but do you think it would make sense to have generic `billing_warning` and `billing_error` fields on an org, and then show those in banners instead? Then we could populate `billing_warning` if it looks like the card will expire within the next 3 months or so, and set `billing_error` if we get a webhook from Stripe that tells us that a payment actually failed or a card has become invalid for some reason.

Reply:
sure that makes sense, however we will need some background jobs to clear out these warnings/errors once they are resolved.

Reply:
or we just rely on webhooks to clear these warnings like payment card updated for expired card warning or payment succeeded for payment failed error. Looks fragile though because webhook delivery may fail or our service may be unavailable at that point etc.

Reply:
ok I see webhooks are retried upto 3 days by stripe but we may need async queue to process the webhooks internally and prevent processing duplicate webhooks

Reply:
should webhook support be added in separate PR ?

Reply:
Yeah we can add it in a separate PR, but I think we should do it sooner rather than later.

Webhooks should be very stable, as you say they are usually retried until ack'ed, but if you worry about inconsistencies you can also have a background job once per day or so that checks and syncs statuses in case a webhook was dropped.

Reply:
Removed checking of card expiry, just keeping check of presence of a payment method so that we can disallow plan change if no payment is present. This is all internal and not exposed in any API so that once we have billing warnings and errors, will just make sure no billing errors present instead of this payment method check.